library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt

```diff
@@ -96,8 +100,24 @@ data class PrivatePreferences(
     var client: Client,
     private val ffiClient: FfiXmtpClient,
 ) {
-    suspend fun syncConsent() {
-        ffiClient.sendSyncRequest(FfiDeviceSyncKind.CONSENT)
+    suspend fun streamHmacKeys(): Flow<Keystore.GetConversationHmacKeysResponse> = callbackFlow {
+        val preferenceCallback = object : FfiPreferenceCallback {
+            override fun onPreferenceUpdate(preference: List<FfiPreferenceUpdate>) {
+                preference.iterator().forEach {
+                    when(it) {
+                        is FfiPreferenceUpdate.Hmac -> it.key
```

Comment:
@codabrink -- To be able to return the `Keystore.GetConversationHmacKeysResponse` I need the epoch and the conversation id.